client/landing/stepper/declarative-flow/newsletter.ts

```diff
@@ -34,17 +35,18 @@ export const newsletter: Flow = {
 		const { setStepProgress } = useDispatch( ONBOARD_STORE );
 		const flowProgress = useFlowProgress( { stepName: _currentStep, flowName: this.name } );
 		setStepProgress( flowProgress );
+		const urlLocale = useLocale();
 
 		function submit( providedDependencies: ProvidedDependencies = {} ) {
 			recordSubmitStep( providedDependencies, '', _currentStep );
-
+			const localeRoute = urlLocale ? `/${ urlLocale }/` : '';
```

Comment:
We don't need to pass `en` when it's English.

```suggestion
			const localeRoute = urlLocale && urlLocale !== 'en' ? `/${ urlLocale }/` : '';
```

Reply:
Indeed, I have this covered on the next commit. Thank you.
🥇